nemo_text_processing/text_normalization/normalize.py

```diff
@@ -329,29 +330,32 @@ def normalize(
         text = text.strip()
         if not text:
             if verbose:
-                print(text)
+                logger.info(text)
             return text
         text = pynini.escape(text)
         tagged_lattice = self.find_tags(text)
         tagged_text = Normalizer.select_tag(tagged_lattice)
         if verbose:
-            print(tagged_text)
+            logger.info(tagged_text)
         self.parser(tagged_text)
         tokens = self.parser.parse()
         split_tokens = self._split_tokens_to_reduce_number_of_permutations(tokens)
         output = ""
         for s in split_tokens:
-            tags_reordered = self.generate_permutations(s)
-            verbalizer_lattice = None
-            for tagged_text in tags_reordered:
-                tagged_text = pynini.escape(tagged_text)
-
-                verbalizer_lattice = self.find_verbalizer(tagged_text)
-                if verbalizer_lattice.num_states() != 0:
-                    break
-            if verbalizer_lattice is None:
-                raise ValueError(f"No permutations were generated from tokens {s}")
-            output += ' ' + Normalizer.select_verbalizer(verbalizer_lattice)
+            try:
+                tags_reordered = self.generate_permutations(s)
+                verbalizer_lattice = None
+                for tagged_text in tags_reordered:
+                    tagged_text = pynini.escape(tagged_text)
+
+                    verbalizer_lattice = self.find_verbalizer(tagged_text)
+                    if verbalizer_lattice.num_states() != 0:
+                        break
+                if verbalizer_lattice is None:
+                    raise ValueError(f"No permutations were generated from tokens {s}")
+                output += ' ' + Normalizer.select_verbalizer(verbalizer_lattice, text)
+            except:
+                raise ValueError("Failed text: " + text)
```

Comment:
let's fall back to the "input" text with an error message but w/o raising the error

Reply:
done